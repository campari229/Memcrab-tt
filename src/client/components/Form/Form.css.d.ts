declare const styles: {
  readonly "form": string;
  readonly "input": string;
  readonly "error": string;
  readonly "button": string;
};
export = styles;

