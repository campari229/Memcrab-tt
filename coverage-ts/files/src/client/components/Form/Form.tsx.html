
  <!DOCTYPE html>
  <html>
    <head>
      <title>Form.tsx</title>
      <link href="https://cdn.jsdelivr.net/npm/semantic-ui@2.4.2/dist/semantic.min.css" type="text/css" rel="stylesheet">
      <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.52.2/codemirror.min.js" type="text/javascript" charset="utf-8"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.52.2/mode/javascript/javascript.min.js" type="text/javascript" charset="utf-8"></script>
<link href="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.52.2/codemirror.min.css" type="text/css" rel="stylesheet">
<script src="..\..\..\..\..\assets\source-file.js" type="text/javascript" charset="utf-8"></script>
<link href="..\..\..\..\..\assets\source-file.css" type="text/css" rel="stylesheet">
    </head>
    <body>
    <div style="margin-top:3em" class="ui container"><h1 class="ui header"><a href="..\..\..\..\..\index.html">TypeScript coverage report</a></h1><table style="margin-top:2em" class="ui celled table"><thead class=""><tr class=""><th class="">Filename</th><th class="">Percent</th><th class="">Threshold</th><th class="">Total</th><th class="">Covered</th><th class="">Uncovered</th></tr></thead><tbody class=""><tr class="positive"><td class="">src\client\components\Form\Form.tsx</td><td class="">100.00%</td><td class="">80%</td><td class="">136</td><td class="">136</td><td class="">0</td></tr></tbody></table><textarea id="editor" readonly="" style="margin-top:3em">import React, { useState, SyntheticEvent } from &#x27;react&#x27;;

import &#x27;./Form.scss&#x27;;

export const Form = () =&gt; {
  const [rowsInputValue, setRowsInputValue] = useState(&#x27;&#x27;);
  const [rowsIsValid, setRowsIsValid] = useState(true);
  const [columnsInputValue, setColumnsInputValue] = useState(&#x27;&#x27;);
  const [columnsIsValid, setColumnsIsValid] = useState(true);
  const [closestInputValue, setClosestInputValue] = useState(&#x27;&#x27;);
  const [closestIsValid, setClosestIsValid] = useState(true);


  const rowsAndColumnsValidation = (inputValue: string) =&gt; {
    if (inputValue &amp;&amp; typeof Number(inputValue) === &#x27;number&#x27;) {
      return true;
    } else {
      return false;
    }
  }

  const closestValidation = (value: string) =&gt; {
    if (rowsInputValue &amp;&amp; columnsInputValue) {
      if (Number(value) &lt;= Number(rowsInputValue) * Number(columnsInputValue)) {
        return true;
      } else {
        return false;
      }
    }
  }

  const makeMatrix = (e: SyntheticEvent) =&gt; {
    e.preventDefault()

    if (rowsAndColumnsValidation(rowsInputValue) &amp;&amp; rowsAndColumnsValidation(columnsInputValue)) {
      if (closestValidation(closestInputValue)) {
        location.assign(`http://localhost:3000/columns=${columnsInputValue}/rows=${rowsInputValue}/closest=${closestInputValue}`);
      } else {
        setClosestIsValid(false);
      }
    } else {
      setRowsIsValid(false);
      setColumnsIsValid(false);
    }
  }

   return (
     &lt;form className=&quot;form&quot; onSubmit={makeMatrix} &gt;
       &lt;h1&gt;Matrix Maker&lt;/h1&gt;
       &lt;div className=&quot;form__input-wrapper&quot;&gt;
          &lt;label className=&quot;form__label&quot;&gt;
            Number of rows
            &lt;input
              type=&quot;text&quot;
              name=&quot;rows&quot;
              className=&quot;form__input&quot;
              onChange={(e) =&gt; setRowsInputValue(e.target.value)}
            /&gt;
          &lt;/label&gt;
          &lt;p className=&quot;form__error&quot; style={rowsIsValid ? { opacity: &#x27;0&#x27; } : {opacity: &#x27;1&#x27;}}&gt;
            Enter a number of rows
          &lt;/p&gt;
       &lt;/div&gt;
       &lt;div className=&quot;form__input-wrapper&quot;&gt;
          &lt;label className=&quot;form__label&quot;&gt;
            Number of columns 
            &lt;input
              type=&quot;text&quot;
              name=&quot;columns&quot;
              className=&quot;form__input&quot;
              onChange={(e) =&gt; setColumnsInputValue(e.target.value)}
            /&gt;
          &lt;/label&gt;
          &lt;p className=&quot;form__error&quot; style={columnsIsValid ? { opacity: &#x27;0&#x27; } : {opacity: &#x27;1&#x27;}}&gt;
            Enter a number of columns
          &lt;/p&gt;
       &lt;/div&gt;
       &lt;div className=&quot;form__input-wrapper&quot;&gt;
          &lt;label className=&quot;form__label&quot;&gt;
          Number of closest cells to shown
            &lt;input
              type=&quot;text&quot;
              id=&quot;closest&quot;
              name=&quot;closest&quot;
              className=&quot;form__input&quot;
              onChange={(e) =&gt; setClosestInputValue(e.target.value)}
            /&gt;
          &lt;/label&gt;
          &lt;p className=&quot;form__error&quot; style={closestIsValid ? { opacity: &#x27;0&#x27; } : {opacity: &#x27;1&#x27;}}&gt;
            Enter a number of closest cells to shown (min: 1, max: columns * rows)
          &lt;/p&gt;
       &lt;/div&gt;
       &lt;button className=&quot;form__button&quot;&gt;Make a matrix&lt;/button&gt;
     &lt;/form&gt;
   )
}
 </textarea><pre id="annotations" style="display:none">[]</pre></div>
    <p class="footer-text">TypeScript Coverage Report generated by <a href="https://github.com/plantain-00/type-coverage">type-coverage</a> and <a href="https://github.com/alexcanessa/typescript-coverage-report">typescript-coverage-report</a> at Wed, 04 Nov 2020 16:03:48 GMT</p>
    </body>
  </html>
  