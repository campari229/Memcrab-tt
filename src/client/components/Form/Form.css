.form {
  width: 40vw;
  margin: 0 auto;
}

.input {
  margin-left: 15px;
  width: 40px;
  height: 30px;

  border: 2px solid black;
  border-radius: 5px;
}

.error {
  color: red;
  margin: 0;
}

.button {
  display: block;
  width: 100px;
  height: 50px;

  margin: 0 auto;

  background-color: black;
  color: white;
  border-radius: 5px;
}
