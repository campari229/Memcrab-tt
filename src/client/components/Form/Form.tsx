import React, { useState, SyntheticEvent } from 'react';

import styles from './Form.css';

export const Form = () => {
  const [rowsInputValue, setRowsInputValue] = useState('');
  const [rowsIsValid, setRowsIsValid] = useState(true);
  const [columnsInputValue, setColumnsInputValue] = useState('');
  const [columnsIsValid, setColumnsIsValid] = useState(true);
  const [closestInputValue, setClosestInputValue] = useState('');
  const [closestIsValid, setClosestIsValid] = useState(true);


  const rowsAndColumnsValidation = (inputValue: string) => {
    if (inputValue && typeof Number(inputValue) === 'number') {
      return true;
    } else {
      return false;
    }
  }

  const closestValidation = (value: string) => {
    if (rowsInputValue && columnsInputValue) {
      if (Number(value) <= Number(rowsInputValue) * Number(columnsInputValue)) {
        return true;
      } else {
        return false;
      }
    }
  }

  const makeMatrix = (e: SyntheticEvent) => {
    e.preventDefault()

    if (rowsAndColumnsValidation(rowsInputValue) && rowsAndColumnsValidation(columnsInputValue)) {
      if (closestValidation(closestInputValue)) {
        location.assign(`http://localhost:3000/columns=${columnsInputValue}/rows=${rowsInputValue}/closest=${closestInputValue}`);
      } else {
        setClosestIsValid(false);
      }
    } else {
      setRowsIsValid(false);
      setColumnsIsValid(false);
    }
  }

   return (
     <form className={styles.form} onSubmit={makeMatrix} >
       <h1>Matrix Maker</h1>
       <div>
          <label>
            Number of rows
            <input
              type="text"
              name="rows"
              className={styles.input}
              onChange={(e) => setRowsInputValue(e.target.value)}
            />
          </label>
          <p className={styles.error} style={rowsIsValid ? { opacity: '0' } : {opacity: '1'}}>
            Enter a number of rows
          </p>
       </div>
       <div>
          <label>
            Number of columns 
            <input
              type="text"
              name="columns"
              className={styles.input}
              onChange={(e) => setColumnsInputValue(e.target.value)}
            />
          </label>
          <p className={styles.error} style={columnsIsValid ? { opacity: '0' } : {opacity: '1'}}>
            Enter a number of columns
          </p>
       </div>
       <div>
          <label>
          Number of closest cells to shown
            <input
              type="text"
              id="closest"
              name="closest"
              className={styles.input}
              onChange={(e) => setClosestInputValue(e.target.value)}
            />
          </label>
          <p className={styles.error} style={closestIsValid ? { opacity: '0' } : {opacity: '1'}}>
            Enter a number of closest cells to shown (min: 1, max: columns * rows)
          </p>
       </div>
       <button className={styles.button}>Make a matrix</button>
     </form>
   )
}
 