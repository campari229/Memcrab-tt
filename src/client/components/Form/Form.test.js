import React from 'react';
import { Form } from './Form';
import TestRender from 'react-test-renderer';

describe('form methods sould work correct', () => {
  let component

  TestRender.act(() => {
    component = TestRender.create(
      <Form />
    )
  })

  it('initial render test', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('rows input test', () => {
    const name = "rows"
    const input = component.root.findByProps({ name: "rows" });
    let event = { target: { name, value: "test" } }

    TestRender.act(() => input.props.onChange(event))

    expect(component.toJSON()).toMatchSnapshot()

    event = { target: { name, value: "" } }
    TestRender.act(() => input.props.onChange(event))
  })

  it('columns input test', () => {
    const name = "columns"
    const input = component.root.findByProps({ name: "columns" });
    let event = { target: { name, value: "test" } }

    TestRender.act(() => input.props.onChange(event))

    expect(component.toJSON()).toMatchSnapshot()

    event = { target: { name, value: "" } }
    TestRender.act(() => input.props.onChange(event))
  })

  it('closest input test', () => {
    const name = "closest"
    const input = component.root.findByProps({ name: "closest" });
    let event = { target: { name, value: "test" } }

    TestRender.act(() => input.props.onChange(event))

    expect(component.toJSON()).toMatchSnapshot()

    event = { target: { name, value: "" } }
    TestRender.act(() => input.props.onChange(event))
  })

  it('test closest input validation', () => {
    const form = component.root.findByProps({ className: 'form' });
    const rowsInput = component.root.findByProps({ name: "rows" });
    const columnsInput = component.root.findByProps({ name: "columns" });
    const closestInput = component.root.findByProps({ name: "closest" });
    let event = { target: { name, value: "2" } }

    TestRender.act(() => rowsInput.props.onChange(event))
    TestRender.act(() => columnsInput.props.onChange(event))
    TestRender.act(() => closestInput.props.onChange({ target: { name, value: "10" } }))

    const withPrevent = { preventDefault: jest.fn() }

    TestRender.act(() => form.props.onSubmit(withPrevent))

    expect(component.root.findByProps({ className: 'form__error closest-error' }).props).toEqual({children: "Enter a number of closest cells to shown (min: 1, max: columns * rows)", className: "form__error closest-error", style: { opacity: "1" } })

    event = { target: { name, value: "" } }

    TestRender.act(() => rowsInput.props.onChange(event))
    TestRender.act(() => columnsInput.props.onChange(event))
    TestRender.act(() => closestInput.props.onChange(event))
  })

  it('form submit test with empty inputs', () => {
    const form = component.root.findByProps({ className: 'form' });

    const withPrevent = { preventDefault: jest.fn() }

    TestRender.act(() => form.props.onSubmit(withPrevent))

    expect(component.root.findByProps({ className: 'form__error row-error' }).props).toEqual({children: "Enter a number of rows", className: "form__error row-error", style: { opacity: "1" } })
  })

  it('form submit test with valid inputs', () => {
    const form = component.root.findByProps({ className: 'form' });
    const rowsInput = component.root.findByProps({ name: "rows" });
    const columnsInput = component.root.findByProps({ name: "columns" });
    const closestInput = component.root.findByProps({ name: "closest" });
    let event = { target: { name, value: "3" } }

    TestRender.act(() => rowsInput.props.onChange(event))
    TestRender.act(() => columnsInput.props.onChange(event))
    TestRender.act(() => closestInput.props.onChange(event))

    // const withPrevent = { preventDefault: jest.fn() }

    TestRender.act(() => form.props.onSubmit({ preventDefault: jest.fn() }))

    expect(component.toJSON()).toMatchSnapshot()
  })
})
