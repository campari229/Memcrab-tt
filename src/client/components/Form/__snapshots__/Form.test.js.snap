// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form methods sould work correct closest input test 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <h1>
    Matrix Maker
  </h1>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of rows
      <input
        className="form__input"
        name="rows"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error row-error"
      style={
        Object {
          "opacity": "0",
        }
      }
    >
      Enter a number of rows
    </p>
  </div>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of columns
      <input
        className="form__input"
        name="columns"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error"
      style={
        Object {
          "opacity": "0",
        }
      }
    >
      Enter a number of columns
    </p>
  </div>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of closest cells to shown
      <input
        className="form__input"
        id="closest"
        name="closest"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error closest-error"
      style={
        Object {
          "opacity": "0",
        }
      }
    >
      Enter a number of closest cells to shown (min: 1, max: columns * rows)
    </p>
  </div>
  <button
    className="form__button"
  >
    Make a matrix
  </button>
</form>
`;

exports[`form methods sould work correct columns input test 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <h1>
    Matrix Maker
  </h1>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of rows
      <input
        className="form__input"
        name="rows"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error row-error"
      style={
        Object {
          "opacity": "0",
        }
      }
    >
      Enter a number of rows
    </p>
  </div>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of columns
      <input
        className="form__input"
        name="columns"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error"
      style={
        Object {
          "opacity": "0",
        }
      }
    >
      Enter a number of columns
    </p>
  </div>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of closest cells to shown
      <input
        className="form__input"
        id="closest"
        name="closest"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error closest-error"
      style={
        Object {
          "opacity": "0",
        }
      }
    >
      Enter a number of closest cells to shown (min: 1, max: columns * rows)
    </p>
  </div>
  <button
    className="form__button"
  >
    Make a matrix
  </button>
</form>
`;

exports[`form methods sould work correct form submit test with valid inputs 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <h1>
    Matrix Maker
  </h1>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of rows
      <input
        className="form__input"
        name="rows"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error row-error"
      style={
        Object {
          "opacity": "1",
        }
      }
    >
      Enter a number of rows
    </p>
  </div>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of columns
      <input
        className="form__input"
        name="columns"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error"
      style={
        Object {
          "opacity": "1",
        }
      }
    >
      Enter a number of columns
    </p>
  </div>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of closest cells to shown
      <input
        className="form__input"
        id="closest"
        name="closest"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error closest-error"
      style={
        Object {
          "opacity": "1",
        }
      }
    >
      Enter a number of closest cells to shown (min: 1, max: columns * rows)
    </p>
  </div>
  <button
    className="form__button"
  >
    Make a matrix
  </button>
</form>
`;

exports[`form methods sould work correct initial render test 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <h1>
    Matrix Maker
  </h1>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of rows
      <input
        className="form__input"
        name="rows"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error row-error"
      style={
        Object {
          "opacity": "0",
        }
      }
    >
      Enter a number of rows
    </p>
  </div>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of columns
      <input
        className="form__input"
        name="columns"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error"
      style={
        Object {
          "opacity": "0",
        }
      }
    >
      Enter a number of columns
    </p>
  </div>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of closest cells to shown
      <input
        className="form__input"
        id="closest"
        name="closest"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error closest-error"
      style={
        Object {
          "opacity": "0",
        }
      }
    >
      Enter a number of closest cells to shown (min: 1, max: columns * rows)
    </p>
  </div>
  <button
    className="form__button"
  >
    Make a matrix
  </button>
</form>
`;

exports[`form methods sould work correct rows input test 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <h1>
    Matrix Maker
  </h1>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of rows
      <input
        className="form__input"
        name="rows"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error row-error"
      style={
        Object {
          "opacity": "0",
        }
      }
    >
      Enter a number of rows
    </p>
  </div>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of columns
      <input
        className="form__input"
        name="columns"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error"
      style={
        Object {
          "opacity": "0",
        }
      }
    >
      Enter a number of columns
    </p>
  </div>
  <div
    className="form__input-wrapper"
  >
    <label
      className="form__label"
    >
      Number of closest cells to shown
      <input
        className="form__input"
        id="closest"
        name="closest"
        onChange={[Function]}
        type="text"
      />
    </label>
    <p
      className="form__error closest-error"
      style={
        Object {
          "opacity": "0",
        }
      }
    >
      Enter a number of closest cells to shown (min: 1, max: columns * rows)
    </p>
  </div>
  <button
    className="form__button"
  >
    Make a matrix
  </button>
</form>
`;
